python_modules/dagster/dagster/_core/definitions/assets.py

```diff
@@ -265,13 +287,27 @@ def __init__(
 
             group_name = normalize_group_name(spec.group_name)
 
-            output_def, _ = node_def.resolve_output_to_origin(output_names_by_key[spec.key], None)
-            metadata = {**output_def.metadata, **(spec.metadata or {})}
+            if node_def is not None:
+                output_def, _ = node_def.resolve_output_to_origin(
+                    output_names_by_key[spec.key], None
+                )
+                node_def_description = node_def.description
+                output_def_metadata = output_def.metadata
+                output_def_description = output_def.description
+                output_def_code_version = output_def.code_version
+                skippable = not output_def.is_required
+            else:
+                node_def_description = None
+                output_def_metadata = {}
+                output_def_description = None
+                output_def_code_version = None
+                skippable = False
+
+            metadata = {**output_def_metadata, **(spec.metadata or {})}
             # We construct description from three sources of truth here. This
             # highly unfortunate. See commentary in @multi_asset's call to dagster_internal_init.
-            description = spec.description or output_def.description or node_def.description
-            code_version = spec.code_version or output_def.code_version
-            skippable = not output_def.is_required
+            description = spec.description or output_def_description or node_def_description
+            code_version = spec.code_version or output_def_code_version
```

Comment:
might be cleaner as a little helper function the returns a new `AssetSpec` to avoid having so many local variables.